extensions/amp-story/1.0/media-pool.js

```diff
@@ -552,7 +552,13 @@ export class MediaPool {
           poolMediaEl,
           new UpdateSourcesTask(this.win_, sources)
         );
-        this.enqueueMediaElementTask_(poolMediaEl, new LoadTask());
+        this.enqueueMediaElementTask_(poolMediaEl, new LoadTask()).then(() => {
```

Comment:
Video elements can be swapped in/out multiple times. Do we need to call this `onVideoLoaded` multiple times?
What is breaking the existing `amp-video` implementation that can't be fixed, so we have to hack things in the media-pool? Do we have a good understanding of the side effects of this hack?

Reply:
The onVideoLoaded provides the media pool with an alternative approach to let amp-video know that the video is loaded, so that it properly sets the state on the amp-video:

https://github.com/ampproject/amphtml/blob/7e21559950ab9bf74650fd6d4ca02d6cc373685f/src/service/video-manager-impl.js#L722-L736

The videoLoaded listens once to LOAD events and starts the percentage tracker, which was not being done before since the videos didn't trigger the LOAD. The exmplanation for why we weren't allowing 
LOAD events to be triggered are because the media pool loads the videos independently from the amp-story tags. More info on that on:

https://github.com/ampproject/amphtml/issues/29938#issuecomment-689195894

Reply:
It's not a "hack" as much as it is a procedure we were missing because we take control of the loading of the videos. AFAIK there are no bad side-effects introduced, as the video-manager-impl only listens for the LOAD once and discards future triggers.

Reply:
I called it a hack because we're not addressing the root cause and just adding some extra code (in another file!) to work around it.

I don't agree with https://github.com/ampproject/amphtml/issues/29938#issuecomment-689195894, the code is still reachable and has nothing to do with the early return. I'll comment directly on the issue.

Reply:
Ok so this is somewhat of the right approach but it should be done in the resetOnDomChange